test/watchos_framework_test.sh

```diff
@@ -0,0 +1,1946 @@
+    minimum_os_version = "2.0",
+    deps = [
+        ":inner_framework",
+        ":outer_framework_lib",
+    ],
+)
+
+objc_library(
+    name = "outer_framework_lib",
+    srcs = [
+        "OuterFramework.h",
+        "OuterFramework.m",
+    ],
+    alwayslink = 1,
+)
+
+apple_dynamic_framework_import(
+    name = "inner_framework",
+    framework_imports = glob(["inner_framework.framework/**"]),
+)
+EOF
+
+  mkdir -p app/inner_framework.framework
+  cp $(rlocation build_bazel_rules_apple/test/testdata/binaries/empty_watchos_dylib_lipobin) \
+      app/inner_framework.framework/inner_framework
+
+  cat > app/inner_framework.framework/Info.plist <<EOF
+Dummy plist
+EOF
+
+  cat > app/inner_framework.framework/resource.txt <<EOF
+Dummy resource
+EOF
+
+  mkdir -p app/inner_framework.framework/Headers
+    
+  cat > app/inner_framework.framework/Headers/fmwk.h <<EOF
+This shouldn't get included
+EOF
+
+  mkdir -p app/inner_framework.framework/Modules
+  cat > app/inner_framework.framework/Headers/module.modulemap <<EOF
+This shouldn't get included
+EOF
+
+  cat > app/Info-Framework.plist <<EOF
+{
+  CFBundleIdentifier = "\${PRODUCT_BUNDLE_IDENTIFIER}";
+  CFBundleName = "\${PRODUCT_NAME}";
+  CFBundlePackageType = "FMWK";
+  CFBundleShortVersionString = "1.0";
+  CFBundleVersion = "1.0";
+}
+EOF
+  cat > app/Info-App.plist <<EOF
+{
+  CFBundleIdentifier = "\${PRODUCT_BUNDLE_IDENTIFIER}";
+  CFBundleName = "\${PRODUCT_NAME}";
+  CFBundlePackageType = "APPL";
+  CFBundleShortVersionString = "1.0";
+  CFBundleVersion = "1.0";
+}
+EOF
+
+  cat > app/Info-Ext.plist <<EOF
+{
+  CFBundleName = "\${PRODUCT_NAME}";
+  CFBundlePackageType = "APPL";
+  CFBundleShortVersionString = "1.0";
+  CFBundleVersion = "1.0";
+  NSExtension = {
+    NSExtensionAttributes = {
+      WKAppBundleIdentifier = "my.bundle.id";
+      NSExtensionPointIdentifier = com.apple.watchkit;
+    };
+    WKExtensionDelegateClassName = "DummyValue";
+  };
+}
+EOF
+
+  cat > app/main.m <<EOF
+int main(int argc, char **argv) {
+  return 0;
+}
+EOF
+
+  cat > app/OuterFramework.h <<EOF
+#ifndef OUTER_FRAMEWORK_OUTER_FRAMEWORK_H_
+#define OUTER_FRAMEWORK_OUTER_FRAMEWORK_H_
+
+void outer();
+
+#endif  // OUTER_FRAMEWORK_OUTER_FRAMEWORK_H_
+EOF
+
+  cat > app/OuterFramework.m <<EOF
+#import <Foundation/Foundation.h>
+
+void outer() {
+  NSLog(@"Outer framework method called\n");
+}
+EOF
+
+  cat > app/InnerFramework.h <<EOF
+#ifndef INNER_FRAMEWORK_INNER_FRAMEWORK_H_
+#define INNER_FRAMEWORK_INNER_FRAMEWORK_H_
+
+void inner();
+
+#endif  // INNER_FRAMEWORK_INNER_FRAMEWORK_H_
+EOF
+
+  cat > app/InnerFramework.m <<EOF
+#import <Foundation/Foundation.h>
+
+void inner() {
+  NSLog(@"Inner framework method called\n");
+}
+EOF
+  
+  do_build watchos //app:app || fail "Should build"
+
+  # Assert that the inner framework was propagated to the application...
+  assert_zip_contains "test-bin/app/app.zip" \
+      "app.app/Frameworks/inner_framework.framework/inner_framework"
```

Comment:
This test is failing as well at the moment:
```
test_framework_depends_on_prebuilt_apple_framework_import FAILED: Archive test-bin/app/app.zip did not contain app.app/Frameworks/inner_framework.framework/inner_framework; contents were: app.app/
app.app/_CodeSignature/
app.app/_CodeSignature/CodeResources
app.app/app
app.app/PlugIns/
app.app/PlugIns/ext.appex/
app.app/PlugIns/ext.appex/_CodeSignature/
app.app/PlugIns/ext.appex/_CodeSignature/CodeResources
app.app/PlugIns/ext.appex/ext
app.app/PlugIns/ext.appex/embedded.mobileprovision
app.app/PlugIns/ext.appex/Info.plist
app.app/_WatchKitStub/
app.app/_WatchKitStub/WK
app.app/Frameworks/
app.app/Frameworks/outer_framework.framework/
app.app/Frameworks/outer_framework.framework/_CodeSignature/
app.app/Frameworks/outer_framework.framework/_CodeSignature/CodeResources
app.app/Frameworks/outer_framework.framework/outer_framework
app.app/Frameworks/outer_framework.framework/Headers/
app.app/Frameworks/outer_framework.framework/Headers/OuterFramework.h
app.app/Frameworks/outer_framework.framework/Info.plist
app.app/embedded.mobileprovision
app.app/Info.plist
app.app/PkgInfo .
```
Maybe this is not relevant since we're not packaging a full IPA?